### ISSUE TITLE
3D OpenGL rendering (Compatibility) is completely black --works normally in 2D

### ISSUE THREAD

**Issue by Say43** [godotengine/godot#79597]
### Godot version

v4.1.1.stable.official [bd6af8e0e]

### System information

Godot v4.1.1.stable - Windows 10.0.19045 - Vulkan (Compatibility) - AMD Radeon HD 7480D () - AMD A4-5300B APU with Radeon(tm) HD Graphics (2 Threads)

### Issue description

When creating a new project, the 3D rendering is completely black, and when moving/looking around, some strange things appear as shown in the following images.
![image](https://github.com/godotengine/godot/assets/125154466/de62d8b6-9204-4831-b27d-8de1094b67c9)
![image](https://github.com/godotengine/godot/assets/125154466/679ec7a7-d426-4ba6-b54b-0ed13d7823b4)

When restarting the engine, the problem is partially resolved. The background/environment appears correctly, but the 3D objects cannot be seen as shown in the following image.
![image](https://github.com/godotengine/godot/assets/125154466/7fee49e1-98cc-43e3-b5cb-a9b1fbae9eaf)

I can confidently say that the problem is not from my device because it was working normally in  v4.0.3.

### Steps to reproduce

Just create a new project and in the renderer option, choose Compatibility (not sure if the problem exists for other people.)

### Minimal reproduction project

N/A

**Comment by Calinou** [comment]
Can you reproduce this if you disable shader caching as instructed in https://github.com/godotengine/godot/issues/79115#issuecomment-1634817174?

**Comment by Say43** [comment]
> Does disabling shader caching as instructed in [#79115 (comment)](https://github.com/godotengine/godot/issues/79115#issuecomment-1634817174) affect this in any way?

No, there was no noticeable difference in the problem after disabling shader caching.
